Subscriber/ApplePayDirectSubscriber.php

```diff
@@ -26,9 +65,62 @@ public static function getSubscribedEvents()
             'Enlight_Controller_Action_PostDispatch_Frontend' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Widget' => 'onFrontendPostDispatch',
             'Enlight_Controller_Action_PostDispatch_Frontend_Checkout' => 'onFrontendCheckoutPostDispatch',
+            'Enlight_Controller_Action_PreDispatch_Frontend_Checkout' => 'onFrontendCheckoutPreDispatch',
+            'Enlight_Controller_Action_PostDispatch_Frontend_Account' => 'onFrontendAccountPostDispatch',
         ];
     }
 
+    /**
+     * @param Enlight_Event_EventArgs $args
+     */
+    public function onFrontendAccountPostDispatch(\Enlight_Event_EventArgs $args)
+    {
+        /** @var Enlight_Controller_Request_RequestHttp $request */
+        $request = $args->get('request');
+
+        if ($request->getActionName() !== 'login') {
```

Comment:
i totally forgot to test the account in the first run.
i did now, but its not working
probably due to this line?

from what i see, it should prevent the user from saving apple pay as default method, but 2 things happen

a) this code isnt triggered at all, because it returns here

b) apple pay and direct are accidentally visible here. we would need to (probably) hide it in chrome like on the checkout payment selection page...but this is also something I can do afterwards...but please let me know
also apple pay direct should always be hidden (but thats also out of this scope which is ok for me)

Reply:
Well, this is just a fallback if you have selected ApplePay as payment method but have not finished the order. On login while switching the browser the default payment method will be reseted.

On the account page for selecting default payment method apple pay could be removed from payment means as done with Apple Pay Direct.

Reply:
so for the test cases:
i use SOFORT and complete the order,
then i start using apple pay in safari, but do not place the order...then i switch to chrome...login..and there it sets it to SOFORT and is immediately visible on the confirm page?

does it really need to happen in the login? cant we move it to somewhere else?
like in account and checkout in those spots?

also yes please remove apple pay direct, and hide apple pay on account page, thank you

Reply:
If we do it in checkout the payment method would change without interaction during the process. Except while logging account is maybe never accessed.

Reply:
Now hidden on account page